### PR TITLE
[9.x] Add `unless` to JsonResource

### PR DESCRIPTION
**Pull request by michaelnabil230**
This PR adds an `unless` method to the JSON resource class for retrieving a value if the given "condition" is (or resolves to) falsy.

### CODE REVIEW

File: tests/Integration/Http/ResourceTest.php [hunk 1]
```diff
@@ -167,6 +168,31 @@ public function testResourcesMayHaveOptionalValues()
         ]);
     }
 
+    public function testResourcesMayDosntHaveOptionalValues()
```
Comment (kevinvanboeckholtz) [inline tests/Integration/Http/ResourceTest.php:171]: Unreadable function name
MayDosntHave
- May Doesn't have
- May Have?
- Doesn't Have?

Comment (michaelnabil230) [reply tests/Integration/Http/ResourceTest.php:171]: Sorry but I’m a little weak in English Can you correct it for me?

Comment (kevinvanboeckholtz) [reply tests/Integration/Http/ResourceTest.php:171]:
```suggestion
    public function testResourcesMayHaveOptionalValues()
```

Comment (kevinvanboeckholtz) [reply tests/Integration/Http/ResourceTest.php:171]:
```suggestion
    public function testResourcesMayHaveOptionalValues()
```

**Pull request merged**